appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/model/util/ModelUtils.java

```diff
@@ -405,7 +399,7 @@ public static boolean isAnnotationNull(Annotation annotation) {
                             return false;
                         } else if (value instanceof Collection && !Collection.class.cast(value).isEmpty()) {
                             return false;
-                        } else if (value instanceof Boolean && Boolean.class.cast(value)) {
+                        } else if (value instanceof Boolean && ((Boolean)value).booleanValue()) {
```

Comment:
I don't particularly like this new syntax, but this comment isn't exactly a blocker so it's up to you

Reply:
While it might not be prettier textual it has the benefit of the compiler helping you out as it can apply type analysis to tell you if this can or cannot work. Using `X.class.cast` which accepts `Object` is valid for any input and the compiler will never complain. This makes this tricky when changes are applied as you don't get compile errors for changes so you tend to overlook needed updates to this style of casting.

The other issue is that it tends to be overused as it looks like a usual method call. Instead of casting once and working type safe from there variables are casted multiple times essentially abandoning static typing for dynamic typing.